core/src/subgraph/instance_manager.rs

```diff
@@ -80,26 +212,29 @@ impl SubgraphInstanceManager {
     }
 
     /// Handle incoming events from subgraph providers.
-    fn handle_subgraph_events<B, S>(
+    fn handle_subgraph_events<B, S, M>(
         logger_factory: LoggerFactory,
         receiver: Receiver<SubgraphAssignmentProviderEvent>,
         stores: HashMap<String, Arc<S>>,
         host_builder: impl RuntimeHostBuilder,
         block_stream_builder: B,
+        metrics_registry: Arc<M>,
     ) where
         S: Store + ChainStore + SubgraphDeploymentStore + EthereumCallCache,
         B: BlockStreamBuilder,
+        M: MetricsRegistry,
     {
         // Subgraph instance shutdown senders
```

Comment:
This comment is only relevant to the `let instances` below, right? Shall we separate that from the metrics variables?